### PR TITLE
Check for legacy devices

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes: #2389

### CODE REVIEW

File: src/js/port_handler.js [hunk 1]
```diff
@@ -242,12 +242,15 @@ PortHandler.selectPort = function(ports) {
...
+            const pathSelect = ports[i].path;
             if (portName.includes('STM') || portName.includes('CP210')) {
-                const pathSelect = ports[i].path;
                 if (OS === 'Windows' || (OS !== 'Windows' && pathSelect.includes('tty'))) {
                     this.portPickerElement.val(pathSelect);
                     console.log(`Porthandler detected device ${portName} on port: ${pathSelect}`);
                 }
+            } else if (portName.includes('tty') && portName.includes("usb")) {
+                this.portPickerElement.val(pathSelect);
+                console.log(`Porthandler detected legacy device ${portName} on port: ${pathSelect}`);
```
Comment (McGiverGim) [inline src/js/port_handler.js:253]: I don't know if we need all this if-else only to show a different log message.

I think is better to unify all of them. We do some extra verification, but I think the overhead is acceptable if the code is better enough. Something like:
```javascript
const boolean isWindows = (OS === 'Windows');
const boolean isTty = pathSelect.includes('tty');
const boolean driverRecognized = portName.includes('STM') || portName.includes('CP210');
const boolean legacyDriverRecognized = portName.includes('usb');
if ((driverRecognized && (isWindows || isTty)) || (legacyDriverRecognized && isTty) {
```
Maybe we can remove the de isWindows and isTty conditions, but you have tested it more.

**Review by McGiverGim — COMMENTED** [review]
Only one suggestion

File: src/js/port_handler.js (hunk 1 above)
Comment (haslinghuis) [reply src/js/port_handler.js:253]: Looks very nice. First implementation had no duplication. But the legacy check introduced some. This seems a clever alternative.

**Review by McGiverGim — DISMISSED** [review]

File: src/js/port_handler.js [hunk 2]
```diff
@@ -242,12 +242,14 @@ PortHandler.selectPort = function(ports) {
...
-                if (OS === 'Windows' || (OS !== 'Windows' && pathSelect.includes('tty'))) {
-                    this.portPickerElement.val(pathSelect);
-                    console.log(`Porthandler detected device ${portName} on port: ${pathSelect}`);
-                }
+            const pathSelect = ports[i].path;
+            const isWindows = (OS === 'Windows');
+            const isTty = pathSelect.includes('tty');
+            const driverRecognized = portName.includes('STM') || portName.includes('CP210');
+            const legacyDriverRecognized = portName.includes('usb');
+            if ((driverRecognized && (isWindows || isTty)) || (legacyDriverRecognized && isTty)) {
```
Comment (mikeller) [inline src/js/port_handler.js:250]: This is very hard to understand now as it mixes different types of conditions in different groupings. How about

```
if (isWindows && driverRecognized || isTty && (driverRecognized || legacyDriverRecognized)) {
```

**Review by mikeller — APPROVED** [review]

**Pull request merged**